addons/fluxcd/resources/deployment/source-controller.yaml

```diff
@@ -1,84 +0,0 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  labels:
-    app.kubernetes.io/instance: flux-system
-    control-plane: controller
-  name: flux-source-controller
-  namespace: flux-system
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app: source-controller
-  strategy:
-    type: Recreate
-  template:
-    metadata:
-      annotations:
-        prometheus.io/port: "8080"
-        prometheus.io/scrape: "true"
-      labels:
-        app: source-controller
-    spec:
-      containers:
-        - args:
-            - --watch-all-namespaces
-            - --log-level=info
-            - --log-encoding=json
-            - --enable-leader-election
-            - --storage-path=/data
-            - --storage-adv-addr=source-controller.$(RUNTIME_NAMESPACE).svc
-          env:
-            - name: RUNTIME_NAMESPACE
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.namespace
-          image: fluxcd/source-controller:v0.25.1
-          imagePullPolicy: IfNotPresent
-          livenessProbe:
-            httpGet:
-              path: /healthz
-              port: healthz
-          name: manager
-          ports:
-            - containerPort: 9090
-              name: http
-            - containerPort: 8080
-              name: http-prom
-            - containerPort: 9440
-              name: healthz
-          readinessProbe:
-            httpGet:
-              path: /
-              port: http
-          resources:
-            limits:
-              cpu: 1000m
-              memory: 1Gi
-            requests:
-              cpu: 50m
-              memory: 64Mi
-          securityContext:
-            allowPrivilegeEscalation: false
-            capabilities:
-              drop:
-              - ALL
-            readOnlyRootFilesystem: true
-            runAsNonRoot: true
-            seccompProfile:
-              type: RuntimeDefault
-          volumeMounts:
-            - mountPath: /data
-              name: data
-            - mountPath: /tmp
-              name: tmp
-      securityContext:
-        fsGroup: 1337
-      serviceAccountName: sa-source-controller
-      terminationGracePeriodSeconds: 10
-      volumes:
-        - emptyDir: { }
-          name: data
-        - emptyDir: { }
-          name: tmp
```

Comment:
did you compare the result by using vela dry-run ? We should check carefully to make sure it's compatible. The fluxcd addon is one of the core addons that be relied on many applications and addons.

Reply:
except security-related stuff, terminationGracePeriodSeconds, and prometheus-related annotations, otherwise they are identical